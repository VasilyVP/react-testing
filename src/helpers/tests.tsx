//import { ReactElement } from 'react'
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom'
import TopLevelRoutes from 'routes/TopLevelRoutes'

type WithRouterProps = {
    //component: ReactElement;
    initialRoute: string;
}
export function renderRoute({ initialRoute }: WithRouterProps = { initialRoute: '/' }) {
    return render(
        <MemoryRouter initialEntries={[initialRoute]}>
            <TopLevelRoutes />
        </MemoryRouter>
    )
}
