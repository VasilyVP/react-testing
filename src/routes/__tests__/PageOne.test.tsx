import { screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { renderRoute } from "helpers/tests"
//import { act } from "react-dom/test-utils"


describe('PageOne', () => {
    it('should match snapshot', () => {
        const { container } = renderRoute({ initialRoute: '/page_one' });
        expect(container).toMatchSnapshot();
    })

    test('clicks + and -', () => {
        renderRoute({ initialRoute: '/page_one' });

        const btnPlus = screen.getByTestId('click+');
        const btnMinus = screen.getByTestId('click-');
        const countEl = screen.getByTestId('count');

        expect(countEl.textContent).toBe('Count: 0');

        userEvent.click(btnPlus);
        expect(countEl.textContent).toBe('Count: 1');

        userEvent.click(btnMinus);
        expect(countEl.textContent).toBe('Count: 0');
    })

    test('back link', () => {
        renderRoute({ initialRoute: '/page_one' });

        const backLink = screen.getByTestId('back_link');
        userEvent.click(backLink);
        const mp = screen.getByTestId('main_page');
        expect(mp).toBeInTheDocument();
    })
})
