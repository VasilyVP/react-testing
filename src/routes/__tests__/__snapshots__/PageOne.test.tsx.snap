// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageOne should match snapshot 1`] = `
<div>
  <div
    data-testid="page_one"
  >
    <h1>
      Page one
    </h1>
    <div
      style="display: flex; flex-direction: column; justify-content: space-between; width: 50px; height: 50px;"
    >
      <button
        data-testid="click+"
      >
        Click +
      </button>
      <button
        data-testid="click-"
      >
        Click -
      </button>
    </div>
    <p
      data-testid="count"
    >
      Count: 
      0
    </p>
    <a
      data-testid="back_link"
      href="/"
    >
      Back
    </a>
  </div>
</div>
`;
